3_javascript/javascript1/week3/scripts.js

```diff
@@ -0,0 +1,153 @@
+// // AND THIS MY USERNAME: SpartanDeBoss
+
+//// Item array removal ////
+const names = [
+  'Peter',
+  'Ahmad',
+  'Yana',
+  'kristina',
+  'Rasmus',
+  'Samuel',
+  'katrine',
+  'Tala',
+];
+const nameToRemove = 'Ahmad';
+
+const indexToRemove = names.indexOf(nameToRemove);
+if (indexToRemove !== -1) {
+  names.splice(indexToRemove, 1);
+}
+
+console.log(names);
+
+///// Series duration of my life /////
+const seriesDurations = [
+  {
+    title: 'The chronicles of Narnia',
+    days: 3,
+    hours: 23,
+    minutes: 15,
+  },
+  {
+    title: 'The Wheel Of Time',
+    days: 4,
+    hours: 5,
+    minutes: 11,
+  },
+  {
+    title: 'Banshee',
+    days: 4,
+    hours: 21,
+    minutes: 55,
+  },
+];
+
+function calculatePercentageOfLife(seriesDurations) {
+  const totalMinutesInALife = 80 * 365 * 24 * 60; // Assuming a life expectancy of 80 years
+
+  let totalMinutesSpentWatching = 0;
+
+  for (const series of seriesDurations) {
+    const { days, hours, minutes } = series;
+    const seriesMinutes = days * 24 * 60 + hours * 60 + minutes;
+    totalMinutesSpentWatching += seriesMinutes;
+  }
+
+  const percentageOfLife =
+    (totalMinutesSpentWatching / totalMinutesInALife) * 100;
+  return percentageOfLife.toFixed(3); // Round to 3 decimal places
+}
+
+function logOutSeriesText() {
+  const totalPercentage = calculatePercentageOfLife(seriesDurations);
+  console.log(`In total that is ${totalPercentage}% of my life`);
+}
+
+logOutSeriesText();
+
+///// Save a note /////
+const notes = [];
+
+function saveNote(content, id) {
+  const note = {
+    content: content,
+    id: id,
+  };
+  notes.push(note);
+}
+
+saveNote('Pick up groceries', 1);
+saveNote('Do laundry', 2);
+
+console.log(notes);
+
+/////// Get a note /////
+function getNote(id) {
+  for (let i = 0; i < notes.length; i++) {
+    if (notes[i].id === id) {
+      return notes[i];
+    }
+  }
+  return null;
+}
+
+const firstNote = getNote(1);
+
+console.log(firstNote);
+
+////// Log out notes /////
+function logOutNotesFormatted() {
+  const notes = [
+    { id: 1, content: 'Pick up groceries' },
+    { id: 2, content: 'Do laundry' },
+  ];
+
+  for (let i = 0; i < notes.length; i++) {
+    console.log(
+      `The note with id: ${notes[i].id}, has the following note text: ${notes[i].content}`
+    );
+  }
+}
+
+logOutNotesFormatted();
+
+////// Adding an activity //////
+let activities = [];
+function addActivity(date, activity, duration) {
+  activities.push;
+  ({
+    date: 'date',
+    activity: 'activity',
+    duration: duration,
+  });
+}
+
+addActivity('23/7-18', 'Youtube', 30);
+
+/*
+activities should now look like this
+[{
+    date: '23/7-18',
+    activity: 'Youtube',
+    duration: 30,
+}]
+*/
+
+function showStatus() {
+  if (activities.length === 0) {
+    return 'Add some activities before calling showStatus';
+  }
+}
+//// Usage limit ////
+if (timeDuration > usageLimit) {
+  return 'You have reached your limit, no more smartphoning for you!';
+}
```

Comment:
Very good job in general. My only observations are that the **travelInformation** exercise is missing and that you should fix the **seriesDurations** exercise.